packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -0,0 +1,294 @@
+// Set by us
+const axiosTimeoutSeconds = Number(getEnvOrFail("AXIOS_TIMEOUT_SECONDS"));
+const bucketName = getEnvOrFail("MEDICAL_DOCUMENTS_BUCKET_NAME");
+// converter config
+const FHIRToCDAConverterUrl = getEnvOrFail("FHIR_TO_CDA_CONVERTER_URL");
+const convertDocLambda = getEnvOrFail("CONVERT_DOC_LAMBDA_NAME");
+const converterKeysTableName = getEnvOrFail("FHIR_T0_CDA_CONVERTER_KEYS_TABLE_NAME");
+
+const lambdaClient = makeLambdaClient(region);
+const s3Client = makeS3Client(region);
+
+export const handler = Sentry.AWSLambda.wrapHandler(
+  async (req: {
+    bundle: Bundle<Resource>;
+    patientId: string;
+    cxId: string;
+    resources?: ResourceTypeForConsolidation[];
+    dateFrom?: string;
+    dateTo?: string;
+    conversionType: string;
+  }) => {
+    const { bundle, patientId, cxId, resources, dateFrom, dateTo, conversionType } = req;
+
+    try {
+      const log = prefixedLog(`patient ${patientId}`);
+      const res = await postToConverter({
+        url: FHIRToCDAConverterUrl,
+        payload: bundle,
+        converterKeysTableName,
+        axiosTimeoutSeconds,
+        log,
+      });
+
+      const formattedXML = formatXML(res.data);
+      const fileName = createFileName({ cxId, patientId, resources, dateFrom, dateTo });
+
+      await s3Client
+        .putObject({
+          Bucket: bucketName,
+          Key: fileName,
+          Body: formattedXML,
+          ContentType: "application/xml",
+        })
+        .promise();
+
+      const convertUrl = await convertDoc({ fileName, conversionType });
+
+      return convertUrl;
+    } catch (err) {
+      console.log(
+        `Error processing bundle for patient: ${patientId} with resources ${resources}; ${err}`
+      );
+      capture.error(err, {
+        extra: {
+          patientId,
+          resources,
+          dateFrom,
+          dateTo,
+          context: lambdaName,
+        },
+      });
+      throw err;
+    }
+  }
+);
+
+const formatXML = (xml: string): string => {
+  const parser = new DOMParser();
+  const document = parser.parseFromString(xml, "text/xml");
+
+  const structuredBody = document.getElementsByTagName("structuredBody")[0];
+  const components = Array.from(structuredBody.getElementsByTagName("component"));
+
+  const problemListComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "Problem list";
+  });
+
+  const treatmentPlanComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "Treatment Plan";
+  });
+
+  const medicationComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "History of Medication use";
+  });
+
+  if (problemListComponent) {
+    const problemListTable = problemListComponent?.getElementsByTagName("table")[0];
```

Comment:
Is this constant of type `any`? See how it can be `undefined`, but the signature of `addIcd10Codes()` requires it to be not undefined. And we call a function on that param on `addIcd10Codes()`, which could lead to runtime error.

I suggest reviewing all usages of this to make sure they are typed accordingly.

Reply:
For some reason it defaults to Element even though it could be undefined i updated the logic so explicetly return undefined if its not true

Reply:
Nevermind the tsconfig suggestion fixes it